cmd/podman/common/specgen.go

```diff
@@ -364,20 +364,20 @@ func FillOutSpecGen(s *specgen.SpecGenerator, c *ContainerCLIOpts, args []string
 	s.WorkDir = workDir
 	entrypoint := []string{}
 	userCommand := []string{}
-	if ep := c.Entrypoint; len(ep) > 0 {
-		// Check if entrypoint specified is json
-		if err := json.Unmarshal([]byte(c.Entrypoint), &entrypoint); err != nil {
-			entrypoint = append(entrypoint, ep)
+	if c.Entrypoint != nil {
+		if ep := *c.Entrypoint; len(ep) > 0 {
+			// Check if entrypoint specified is json
+			if err := json.Unmarshal([]byte(*c.Entrypoint), &entrypoint); err != nil {
+				entrypoint = append(entrypoint, ep)
+			}
 		}
+		s.Entrypoint = entrypoint
 	}
-
 	var command []string
 
-	s.Entrypoint = entrypoint
-
 	// Build the command
 	// If we have an entry point, it goes first
-	if len(entrypoint) > 0 {
+	if c.Entrypoint != nil {
```

Comment:
Why are we building the command here? It seems like we should be unconditionally setting s.Command to userCommand?

Reply:
I've needed it to differentiate between an empty entrypoint and an unspecified entrypoint.

If `entrypoint == nil` then it wasn't specified.